sdk/src/client/api/block_builder/transaction_builder/mod.rs

```diff
@@ -486,30 +485,28 @@ impl TransactionBuilder {
     fn select_input(&mut self, input: InputSigningData) -> Result<Option<&Output>, TransactionBuilderError> {
         log::debug!("Selecting input {:?}", input.output_id());
 
-        let mut added_output = None;
+        let mut added_output = false;
         if let Some(output) = self.transition_input(&input)? {
             // No need to check for `outputs_requirements` because
             // - the sender feature doesn't need to be verified as it has been removed
             // - the issuer feature doesn't need to be verified as the chain is not new
             // - input doesn't need to be checked for as we just transitioned it
             // - foundry account requirement should have been met already by a prior `required_account_nft_addresses`
             self.added_outputs.push(output);
-            added_output = self.added_outputs.last();
+            added_output = true;
         }
 
         if let Some(requirement) = self.required_account_nft_addresses(&input)? {
             log::debug!("Adding {requirement:?} from input {:?}", input.output_id());
             self.requirements.push(requirement);
         }
 
-        self.selected_inputs.push(input);
+        // New input may need context inputs
+        self.fulfill_context_inputs_requirements(&input);
```

Comment:
What about the output requirements if we added one?

Reply:
There can't be any new requirements for them or can there?

Reply:
I don't know, maybe there can

Reply:
From what I see we only need it for delegation outputs and since we never create them if not provided (there we check it already) we don't need any additional check